addons/DistancePlus/DistancePlus.lua

```diff
@@ -0,0 +1,392 @@
+  abilitylistrecast = windower.ffxi.get_ability_recasts() 
+  abilitylist = windower.ffxi.get_abilities().job_abilities
+  list = 'Abilities:\n'
+  for key,ability in pairs(abilitylist) do
+    ability_en = res.job_abilities[ability].en
+	ability_type = res.job_abilities[ability].type
+	ability_targets = res.job_abilities[ability].targets
+	ability_distance = jadistance[ability].distance or 0
+	if distance and ability_en and (ability_type == 'JobAbility' or ability_type == 'PetCommand' or ability_type == 'BloodPactRage' or ability_type == 'BloodPactWard' or ability_type == 'Monster' or ability_type == 'Step') and ability_en ~= "Flourishes II" then 
+		if ability_targets.Self ~= true then
+			if distance < (ability_distance + s.model_size + t.model_size) and distance ~= 0 then 
+				list = list..'\\cs(0,255,0)'..ability_en..'\\cs(255,255,255)'..'\n'
+			else
+				list = list..'\\cs(255,255,255)'..ability_en..'\n'
+			end
+		--- too much crap on screen!!!
+--		elseif ability_targets.Self == true and (ability_type == 'Monster' or ability_type == 'PetCommand') and master_pet_distance then
+--			if master_pet_distance < (4 + s.model_size + t.model_size) and distance ~= 0 then 
+--				list = list..'\\cs(0,255,0)'..ability_en..'\\cs(255,255,255)'..'\n'
+--			else
+--				list = list..'\\cs(255,255,255)'..ability_en..'\n'
+--			end
+		end
+	end
+  end
+  abilities.value = list
+  abilities:visible(showabilities)
+end
+
+function check_job()
+    self = windower.ffxi.get_player()
+	windower.add_to_chat(8,'*****DP Job Selection:'..self.main_job..'*****')
+	if self.main_job == 'RDM' or self.main_job == 'BLM' or self.main_job == 'GEO' or self.main_job == 'SCH' or self.main_job == 'WHM' or self.main_job == 'BRD'  then
+		option = "Magic"
+		windower.add_to_chat(8,'Mode: Magic.')
+		windower.add_to_chat(8,' White = Can not cast.')
+		windower.add_to_chat(8,' Green = Casting Range')
+		MaxDistance = 20	 
+	elseif self.main_job == 'COR' then
+		windower.add_to_chat(8,'Mode: Gun.')
+		windower.add_to_chat(8,' White  = Can not shoot.')
+		windower.add_to_chat(8,' Yellow = Ranged Attack Capable (No Buff)')
+		windower.add_to_chat(8,' Green  = Shoots Squarely (Good)')
+		windower.add_to_chat(8,' Blue   = True Shot (Best)')
+		option = "Gun"
+		MaxDistance = 25
+	elseif self.main_job == 'RNG' then
+		windower.add_to_chat(8,'RANGER should do //dp Bow, //dp XBow, or //dp Gun')
+		windower.add_to_chat(8,'Mode: Default.')
+	    option = "Default"
+	    MaxDistance = 25
+	elseif self.main_job == 'NIN' then
+		option = "Ninjutsu"
+		windower.add_to_chat(8,'Mode: Ninjutsu.')
+		windower.add_to_chat(8,' White = Can not cast.')
+		windower.add_to_chat(8,' Green = Casting Range')
+	else
+		windower.add_to_chat(8,'Mode: Default.')
+	    option = "Default"
+	    MaxDistance = 25
+	end
+end
+
+
+windower.register_event('prerender', function()
+    t = windower.ffxi.get_mob_by_index(windower.ffxi.get_player().target_index or 0)
```

Comment:
`windower.ffxi.get_player` is an expensive function, and you call it twice in each `prerender` event, despite having an unused `self` variable you could use. And you don't even need that, should rewrite it to:
```lua
    t = windower.ffxi.get_mob_by_target('t') or windower.ffxi.get_mob_by_target('st')`
    s = windower.ffxi.get_mob_by_target('me')
    pet = windower.ffxi.get_mob_by_target('pet')
```

The last call you make to get the pet is especially puzzling. You get the pet mob, then get its index and get the same object again by calling another expensive function. You should keep in mind that none of this is free and you can save yourself a lot of effort when doing it right.